agent/tsunami/tsunami.py

```diff
@@ -72,35 +77,44 @@ def _start_scan(self, target, output_file: str):
 
         subprocess.run(tsunami_command, encoding='utf-8', stdout=subprocess.DEVNULL, check=True)
 
-    def _parse_result(self, output_file):
+    def _parse_result(self, output_file: Optional[Any]) -> Dict[str, Any]:
         """After the scan is done, parse the output json file into a dict of the scan Findings.
         returns:
             - scan results.
         """
-        logger.info('scan is done, parsing the results from %s.', output_file.name)
-        tsunami_result = json.load(output_file)
-        json_result = {
+        json_result: Dict[str, Any] = {
             'vulnerabilities': []
         }
-        if 'SUCCEEDED' in tsunami_result['scanStatus'] and 'scanFindings' in tsunami_result.keys():
-            json_result['status'] = 'success'
-            logger.info('scan status: SUCCEEDED')
-            for vul in tsunami_result['scanFindings']:
-                json_result['vulnerabilities'].append(vul)
-        else:
-            json_result['status'] = 'failed'
+        if output_file is not None:
+            logger.info('scan is done, parsing the results from %s.', output_file.name)
+            tsunami_result = json.load(output_file)
+
+            if 'SUCCEEDED' in tsunami_result['scanStatus'] and 'scanFindings' in tsunami_result.keys():
+                json_result['status'] = 'success'
+                logger.info('scan status: SUCCEEDED')
+                for vul in tsunami_result['scanFindings']:
+                    json_result['vulnerabilities'].append(vul)
+            else:
+                json_result['status'] = 'failed'
         return json_result
 
-    def scan(self, target: Target):
+    def scan(self, target: Target) -> Dict[str, Any]:
         """Start a scan, wait for the scan results and clean the scan output.
 
            returns:
             - Scan results from tsunami.
         """
-        self._start_scan(target, self._output_file.name)
-        findings = self._parse_result(self._output_file)
-        return findings
+        try:
+            if self._output_file is not None:
+                self._start_scan(target, self._output_file.name)
+                findings = self._parse_result(self._output_file)
+                return findings
+            else:
+                return {}
+        except func_timeout.exceptions.FunctionTimedOut:
+            return {}
 
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        self._output_file.close()
+    def __exit__(self, exc_type: str, exc_val: str, exc_tb: str) -> Any:
```

Comment:
I think the types should be:
exc_type: type[BaseException] | None,
exc_val: BaseException | None,
exc_tb: TracebackType | None,